### ISSUE TITLE
Feature Request: gitversion honor --merged and --no-merged flags

### ISSUE THREAD

**Issue by scr-oath** [screwdriver-cd/screwdriver#2777]
TL;DR - some teams wish to keep major and/or minor branches alive with separate "patch" bumps; would like a feature to pull the "latest tag" from the branch only.

Some teams may wish to keep several version branches alive to support customers pinned to major or minor by making patches to those version branches.

[This StackOverflow post](https://stackoverflow.com/questions/2381665/list-tags-contained-by-a-branch) suggests that git now supports `--merged` and `--no-merged` flags to govern how tags are found.  Perhaps this could be used to allow each branch to have a different "latest version" to be bumped.

**Comment by scr-oath** [comment]
Well… if there are multiple screwdriver pipelines pointing at different branches, the desire would be to use the tags only on that branch - as I understand it `git tag` lists _ALL_ tags, whereas `git tag --merged` lists only tags that are in the current branch…

Taking a stab at this with a local repo… where the main has two tags on 0.0 and there are two branches v0.0 and v0.1, you can see that, if we can pass the `--merged` flag down to the `git tag` call, branches can maintain their own view of the world and bump independently:

```bash
[scr@R9459YDHVQ]$ git symbolic-ref --short HEAD
main
[scr@R9459YDHVQ]$ git tag --merged
v0.0.0
v0.0.1
```

```bash
[scr@R9459YDHVQ]$ git symbolic-ref --short HEAD
v0.1
[scr@R9459YDHVQ]$ git tag --merged
v0.0.0
v0.0.1
v0.1.0
```

```bash
[scr@R9459YDHVQ]$ git symbolic-ref --short HEAD
v0.0
[scr@R9459YDHVQ]$ git tag --merged
v0.0.0
v0.0.1
v0.0.2
```

**Comment by jithine** [comment]
fixed with https://github.com/screwdriver-cd/gitversion/releases/tag/v1.1.9
